utils/utils.go

```diff
@@ -574,3 +574,48 @@ func downloadProgressSetup(ctx context.Context,
 	}()
 	return writer, cancel
 }
+
+// AtomicCopy implements a best effort to atomically copy the file at src to
+// dst. It does this by copying first to a temporary file in the same directory
+// as dst, then renaming that file to the final expected path.
```

Comment:
[q] Why is an atomic copy necessary? Do you expect there to be some other, concurrent creator of an identically-named agent binary in cache? Or, is this just a best principles kind of thing?

Reply:
Just a best practices thing. This is how a lot of system management programs like Chef handle managed files because it prevents leaving the file in a partially copied state if the program or OS crashes mid copy and works across file systems.